gensim/models/wrappers/fasttext.py

```diff
@@ -270,26 +278,34 @@ def load_model_params(self, file_handle):
         self.wv.max_n = maxn
         self.sample = t
 
-    def load_dict(self, file_handle):
-        (vocab_size, nwords, _) = self.struct_unpack(file_handle, '@3i')
+    def load_dict(self, file_handle, encoding='utf8'):
+        vocab_size, nwords, _ = self.struct_unpack(file_handle, '@3i')
         # Vocab stored by [Dictionary::save](https://github.com/facebookresearch/fastText/blob/master/src/dictionary.cc)
         assert len(self.wv.vocab) == nwords, 'mismatch between vocab sizes'
-        assert len(self.wv.vocab) == vocab_size, 'mismatch between vocab sizes'
-        ntokens, = self.struct_unpack(file_handle, '@q')
+        if len(self.wv.vocab) != vocab_size:
+            logger.warnings("If you are loading any model other than pretrained vector wiki.fr, ")
+            logger.warnings("Please report to gensim or fastText.")
+        ntokens= self.struct_unpack(file_handle, '@1q')
+        if self.new_format:
+            pruneidx_size = self.struct_unpack(file_handle, '@q')
```

Comment:
`pruneidx_size` seems to be used in the original FastText code to load dictionaries -
<pre>
void Dictionary::load(std::istream& in) {
  words_.clear();
  std::fill(word2int_.begin(), word2int_.end(), -1);
  in.read((char*) &size_, sizeof(int32_t));
  in.read((char*) &nwords_, sizeof(int32_t));
  in.read((char*) &nlabels_, sizeof(int32_t));
  in.read((char*) &ntokens_, sizeof(int64_t));
  in.read((char*) &pruneidx_size_, sizeof(int64_t));
  for (int32_t i = 0; i < size_; i++) {
    char c;
    entry e;
    while ((c = in.get()) != 0) {
      e.word.push_back(c);
    }
    in.read((char*) &e.count, sizeof(int64_t));
    in.read((char*) &e.type, sizeof(entry_type));
    words_.push_back(e);
    word2int_[find(e.word)] = i;
  }
  pruneidx_.clear();
<b>
  for (int32_t i = 0; i < pruneidx_size_; i++) {
    int32_t first;
    int32_t second;
    in.read((char*) &first, sizeof(int32_t));
    in.read((char*) &second, sizeof(int32_t));
    pruneidx_[first] = second;
  }
</b>
  initTableDiscard();
  initNgrams();
}
</pre>
I'm not sure if it's present in the models we're loading or not though?